src/transformers/models/oneformer/configuration_oneformer.py

```diff
@@ -0,0 +1,205 @@
+            Dictionary containing general configuration like backbone_type, loss weights, number of classes, etc.
+        backbone_config (`dict`, *optional*, defaults to `swin-tiny-patch4-window7-224`)
+            Dictionary containing configuration for the backbone module like patch size, num_heads, etc.
+        text_encoder_config (`dict`, *optional*, defaults to a dictionary with the following keys)
+            Dictionary containing configuration for the text-mapper module and task encoder like sequence length,
+            number of linear layers in MLP, etc.
+        decoder_config (`dict`, *optional*, defaults to a dictionary with the following keys)
+            Dictionary containing configuration for the text-mapper module and task encoder like sequence length,
+            number of linear layers in MLP, etc.
+
+    Raises:
+        `ValueError`:
+            Raised if the backbone model type selected is not in `["swin", "dinat"]`
+    Examples:
+    ```python
+    >>> from transformers import OneFormerConfig, OneFormerModel
+
+    >>> # Initializing a OneFormer shi-labs/oneformer_ade20k_swin_tiny configuration
+    >>> configuration = OneFormerConfig()
+    >>> # Initializing a model (with random weights) from the shi-labs/oneformer_ade20k_swin_tiny style configuration
+    >>> model = OneFormerModel(configuration)
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+    """
+    model_type = "oneformer"
+    backbones_supported = ["swin", "dinat"]
+
+    def __init__(
+        self,
+        general_config: Optional[Dict] = None,
+        backbone_config: Optional[Dict] = None,
+        text_encoder_config: Optional[Dict] = None,
+        decoder_config: Optional[Dict] = None,
+        **kwargs,
+    ):
+        cfgs = self._setup_cfg(general_config, backbone_config, text_encoder_config, decoder_config)
+
+        general_config, backbone_config, text_encoder_config, decoder_config = cfgs
+
+        self.general_config = general_config
+        self.backbone_config = backbone_config
+        self.text_encoder_config = text_encoder_config
+        self.decoder_config = decoder_config
+
+        self.hidden_size = self.decoder_config["hidden_dim"]
+        self.num_attention_heads = self.decoder_config["num_heads"]
+        self.num_hidden_layers = self.decoder_config["decoder_layers"]
+        self.init_std = self.general_config["init_std"]
+        self.init_xavier_std = self.general_config["init_xavier_std"]
+
+        super().__init__(**kwargs)
+
+    def _setup_cfg(
+        self,
+        general_config: Optional[Dict] = None,
+        backbone_config: Optional[SwinConfig] = None,
+        text_encoder_config: Optional[Dict] = None,
+        decoder_config: Optional[Dict] = None,
+    ) -> Dict[str, any]:
+        if general_config is None:
+            general_config = {}
+            general_config["ignore_value"] = 255
+            general_config["num_classes"] = 150
```

Comment:
The "number of classes" argument is normally not defined for Transformer-based models.

All models use `config.num_labels` for this purpose (along with `config.id2label` and `config.label2id`)